File: Atspm/WebUI/src/features/locations/types/DetectorDto.ts
// #region license
// Copyright 2024 Utah Departement of Transportation
// for WebUI - DetectorDto.ts
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//http://www.apache.org/licenses/LICENSE-2.
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// #endregion
import { directionType } from './DirectionType'

interface DetectorsDto {
  id: number
  detectionChannel: number
  direction: directionType
  phase: number
  permPhase: number
  overlap: boolean
  enabled: boolean
  detectionTypes: string
  detectionHardware: string
  latencyCorrection: string
  movementType: string
  laneNumber: number
  laneType: string
  distanceFromStopBar: number
  decisionPoint: string
  movementDelay: string
  minSpeedFilter: number
  comment: string
  detectorComments: Comment[]
}
interface Comment {
  comment: string
  timeStamp: string
  detectorId: number
  id: number
}

export default DetectorsDto

File: Atspm/ConfigApi/DTO/DetectorDto.cs
#region license
// Copyright 2025 Utah Departement of Transportation
// for ConfigApi - Utah.Udot.ATSPM.ConfigApi.DTO/DetectorDto.cs
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
// http://www.apache.org/licenses/LICENSE-2.
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
#endregion

using Utah.Udot.Atspm.Data.Enums;

namespace Utah.Udot.ATSPM.ConfigApi.DTO
{
    public class DetectorDto
    {
        public int? Id { get; set; }
        public string DectectorIdentifier { get; set; }
        public int DetectorChannel { get; set; }
        public int? DistanceFromStopBar { get; set; }
        public int? MinSpeedFilter { get; set; }
        public DateTime DateAdded { get; set; }
        public DateTime? DateDisabled { get; set; }
        public int? LaneNumber { get; set; }
        public MovementTypes MovementType { get; set; }
        public LaneTypes LaneType { get; set; }
        public DetectionHardwareTypes DetectionHardware { get; set; }
        public int? DecisionPoint { get; set; }
        public int? MovementDelay { get; set; }
        public double LatencyCorrection { get; set; }
        public int? ApproachId { get; set; }
        public ICollection<DetectionTypeDto> DetectionTypes { get; set; }
    }
}
